dapp/src/components/TransactionListener.js

```diff
@@ -188,29 +184,33 @@ class TransactionListener extends Component {
       s.transactions = storageTransactions
     })
 
-    // need to call it 1 frame later so that `transactions` props get populated
-    setTimeout(async () => {
-      await this.observeTransactions(storageTransactions)
-    }, 1)
+    setTimeout(async () => await observeTransactions(storageTransactions), 1)
   }
 
-  save(transactions) {
-    localStorage.setItem(
-      this.localStorageId(this.props.account),
-      JSON.stringify(transactions)
-    )
+  /**
+   * Save transaction data to local storage.
+   */
+  const save = (transactions) => {
+    localStorage.setItem(localStorageId(account), JSON.stringify(transactions))
   }
 
-  async observeTransaction(transaction) {
+  const observeTransaction = async (transaction) => {
+    const observableHash = transaction.safeData
```

Comment:
would it be safer to check for `isSafe` here: 
```
const observableHash = transaction.isSafe
  ? transaction.safeData && transaction.safeData.txHash (otherwise SDK call failed I guess?)
  : transaction.hash
```

Reply:
Hmm yea if the SDK call fails observableHash will end up being false which will cause it to error out. The way it is written now transaction.hash will at least exist (it'll be wrong and the listener won't update properly but at least it is preferable to erroring out). We need to address this.